### PR TITLE
Virtual kubelet metrics API

### PR DESCRIPTION
**Pull request by mlavacca**
# Description

This PR introduces the metrics API implementation in the virtual kubelet. The virtual kubelet exposes a new endpoint for retrieving the stats of the offloaded pods running in the remote cluster.

* the `vk` iterates over all the namespaces currently natted
* all the pods in each natted namespace in the remote cluster having a specific label (a reference to the offloading virtual kubelet) are listed from the metrics API server
* the data structure used by the kubelet is built upon the remote data and is retrieved to the caller.

### CODE REVIEW

**Comment by palexster** [comment]
ref. #367

File: internal/virtualKubelet/node/api/server.go [hunk 1]
```diff
@@ -47,8 +50,22 @@ func PodHandler(p PodHandlerConfig, debug bool) http.Handler {
...
 	r.HandleFunc("/containerLogs/{namespace}/{pod}/{container}", HandleContainerLogs(p.GetContainerLogs)).Methods("GET")
-	r.HandleFunc("/exec/{namespace}/{pod}/{container}", HandleContainerExec(p.RunInContainer)).Methods("POST")
+	r.HandleFunc(
+		"/exec/{namespace}/{pod}/{container}",
+		HandleContainerExec(
+			p.RunInContainer,
+		),
+	).Methods("POST", "GET")
+
+	if p.GetStatsSummary != nil {
```
Comment (palexster) [inline internal/virtualKubelet/node/api/server.go:63]: Why do we have this if? We are checking if the remote cluster has metrics enabled?

Comment (mlavacca) [reply internal/virtualKubelet/node/api/server.go:63]: This is for being sure the metrics handler has been set. Actually, this if is always true in our implementation, I can remove it.

**Review by palexster — APPROVED** [review]

**Review by alacuku — APPROVED** [review]

**Pull request merged**